### PR TITLE
get user in the splash and navigate to home screen or auth screen if …

### PR DESCRIPTION
**Pull request by ShonOhana**
…the user is logged in and also navigate to home screen if login or register is succeed

### CODE REVIEW

File: app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/AuthScreen.kt [hunk 1]
```diff
@@ -35,11 +39,19 @@ fun AuthScreen(
         when (currentScreen) {
             AuthScreenState.Login -> LoginScreen(
                 viewModel = authViewModel,
-                onRegisterClicked = { currentScreen = AuthScreenState.Register }
+                onRegisterClicked = { currentScreen = AuthScreenState.Register },
```
Comment (RazCoH) [inline app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/AuthScreen.kt:42]: We should add else logic , for now maybe just prompt toast message. In the future we will show error dialog.

File: app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/AuthScreen.kt [hunk 2]
```diff
@@ -35,11 +39,19 @@ fun AuthScreen(
...
+                    if (succeess && exception == null)
+                        navigator.navigate(MainScreens.Home)
+                }
             )
             AuthScreenState.Register -> RegisterScreen(
                 viewModel = authViewModel,
-                onLoginClicked = { currentScreen = AuthScreenState.Login }
+                onLoginClicked = { currentScreen = AuthScreenState.Login },
+                isRegisterSucceed = { succeess, exception ->
+                    if (succeess && exception == null)
```
Comment (RazCoH) [inline app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/AuthScreen.kt:52]: Also here. We should add else case

File: app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/register/AuthViewModel.kt [hunk 3]
```diff
@@ -89,20 +93,34 @@ class AuthViewModel(
                         if (success){
                             _signupData.value = signupData.copy(authError = false)
                             uiAuthState.value = UIState.Success(true)
+                            successCallback(true,null)
                         }else {
                             uiAuthState.value = UIState.Error(Constants.UNKNOWN_EXCEPTION)
```
Comment (RazCoH) [inline app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/register/AuthViewModel.kt:98]: It is really unknown exception?

**Review by RazCoH — CHANGES_REQUESTED** [review]
Just add the else blocks with something temporary

File: app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/AuthScreen.kt (hunk 1 above)
Comment (ShonOhana) [reply app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/AuthScreen.kt:42]: if there is an exception the uiAuthState will show the red error text above the login button in login and register according to the figma so meanwhile I think its ok

File: app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/AuthScreen.kt (hunk 2 above)
Comment (ShonOhana) [reply app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/AuthScreen.kt:52]: if there is an exception the uiAuthState will show the red error text above the login button in login and register according to the figma so meanwhile I think its ok

File: app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/register/AuthViewModel.kt (hunk 3 above)
Comment (ShonOhana) [reply app/src/main/java/com/example/composetemplate/presentation/screens/entry_screens/register/AuthViewModel.kt:98]: the call returning boolean if the response code in the 200 zone.. if its not success I think its 401. there is extension you did to handle it in the client networking so its should work autumatcly no ? and if its not 401 then yes its enknown exception

**Pull request merged**